### PR TITLE
Updated Upstream (Bukkit/CraftBukkit/Spigot)

### LINKED ISSUE #5843: Generate Tree Block Change Delegate
### Expected behavior

The BlockChangeDelegate provided in the method World#generateTree(Location, TreeType, BlockChangeDelegate) does something

### Observed/Actual behavior

The BlockChangeDelegate provided in the method World#generateTree(Location, TreeType, BlockChangeDelegate)

### Steps/models to reproduce

Look at the source code for World#generateTree(Location, TreeType, BlockChangeDelegate)

### Plugin list

N/A

### Paper version

Latest (not fixed in any recent patches)

### Agreements

- [X] I am running the latest version of Paper available from https://papermc.io/downloads.
- [X] I have searched for and ensured there isn't already an open issue regarding this.
- [X] My version of Minecraft is supported by Paper.

### Other

I would suggest possibly changing the functionality of the BlockChangeDelegate to instead accept blocks after the tree is generated - this would be a lot more user friendly than dealing with block data. However, what do I know; do whatever you smart people think is best!

### PR DESCRIPTION
**Pull request by kennytv**
Upstream has released updates that appear to apply and compile correctly.
This update has not been tested by PaperMC and as with ANY update, please do your own testing

Bukkit Changes:
c9a46ebf #653: Add World#spawn with randomizeData parameter
e49c2e3a Damageable should extend ItemMeta
01ff04f4 SPIGOT-5880, SPIGOT-5567: New ChunkGenerator API

CraftBukkit Changes:
7c8bbcbe SPIGOT-6716: Preserve the order of stored enchantments of enchanted books.
18027d02 #914: Add World#spawn with randomizeData parameter
3cad0316 SPIGOT-6714: Don't fire PlayerBucketEvent when empty
8c6d60cf Fix server crash with BlockPopulator when entities are at a negative chunk border
4f6bcc84 SPIGOT-5880, SPIGOT-5567: New ChunkGenerator API

Spigot Changes:
b166a49b Rebuild patches

### CODE REVIEW

**Comment by kennytv** [comment]
This giant turd + a new patch patch is brought to you by Spigot Ltd.

**Comment by NoahvdAa** [comment]
[This](https://github.com/PaperMC/Paper/blob/master/patches/server/0747-Do-not-fire-PlayerBucketFillEvent-when-no-fluid.patch) patch can be deleted, as it was [fixed upstream](https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/pull-requests/917/overview).

**Comment by Machine-Maker** [comment]
Actually it can't be deleted. See the [very first file in the diff](https://github.com/PaperMC/Paper/pull/6457/files#diff-fe5e4b199830bee6520fdaa6be433f8ca16a11540a97b4c2f9aea017ac5d6467). It removes upstreams fix. so we have to add it back.

**Comment by electronicboy** [comment]
The commit upstream was bad, hence the change there was manually reverted

**Comment by NoahvdAa** [comment]
My broken upstream commit [has been rebuilt](https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/commits/15792f0de16e65649c931a3c911ab35b4994abd3).

**Review by electronicboy — APPROVED** [review]

File: patches/server/0418-Load-Chunks-for-Login-Asynchronously.patch [hunk 1]
```diff
@@ -243,16 +243,3 @@ index 9609579e6d61c3975740d07a394680aeef952516..6b23b9389ff92ae8016d4adb289ac2a0
...
-diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
-index 079e8f0f7ec5c3566b6778f5d518061a08699872..ebf7a58c6af5f5573d97c730c50c69a9f8140a6d 100644
---- a/src/main/java/net/minecraft/world/entity/Entity.java
-+++ b/src/main/java/net/minecraft/world/entity/Entity.java
-@@ -1513,7 +1513,7 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource, n
-         this.yo = y;
-         this.zo = d4;
-         this.setPos(d3, y, d4);
--        this.level.getChunk((int) Math.floor(this.getX()) >> 4, (int) Math.floor(this.getZ()) >> 4); // CraftBukkit
-+        if (valid) this.level.getChunk((int) Math.floor(this.getX()) >> 4, (int) Math.floor(this.getZ()) >> 4); // CraftBukkit // Paper
```
Comment (Proximyst) [inline patches/server/0418-Load-Chunks-for-Login-Asynchronously.patch:255]: Why?

**Review by Proximyst — APPROVED** [review]

**Pull request merged**